### PR TITLE
Fix for issue #9465

### PR DESCRIPTION
**Pull request by NuLL3rr0r**
$SHELL returns the correct shell path on *nix system

### CODE REVIEW

**Comment by stanhu** [comment]
Seems reasonable to me. @jacobvosmaer, any reason this shouldn't go in?

**Comment by jacobvosmaer** [comment]
$SHELL returns the shell of the current user. It is not uncommon for root to use a different shell than normal users, and commands like `su` and `sudo` have complicated rules/options for either changing or not changing $SHELL when switching user. I don't think relying on $SHELL in an init script is a good idea.

I know that systems like FreeBSD do not have /bin/bash, even if you install bash it goes into another directory.

If `shell_dir=/bin/bash` is a problem on some systems, I think the solution should be to make the init script work with `sh` (plain POSIX shell) rather than relying on Bash.

TL;DR use /etc/default/gitlab if your bash is not in /bin. The 'real' solution is to use /bin/sh in the init script instead of Bash.

**Comment by jacobvosmaer** [comment]
You can just write this in /etc/default/gitlab:

```
shell_path=/usr/local/bin/bash
```

**Comment by NuLL3rr0r** [comment]
Using /bin/sh seems a good idea to me. But there is no /etc/default/gitlab on my system since I did installed it from source. And I never saw anything about it on the installation docs. But I'll check it out.
By the way on *BSD operating system anything outside base system goes into /usr/local/ e.g. /usr/local/bin for binaries and /usr/local/etc/ for configuration files or /usr/local/etc/rc.d/ for init scripts. So I suppose it must be /usr/local/etc/default/gitlab on BSD family of operating systems.

**Comment by jacobvosmaer** [comment]
@NuLL3rr0r see this comment for the /etc/default/gitlab file: https://github.com/gitlabhq/gitlabhq/blob/82f0d3e655c48fc67adea88af681d0b5d81810de/lib/support/init.d/gitlab#L20

There is also a template in https://github.com/gitlabhq/gitlabhq/blob/master/lib/support/init.d/gitlab.default.example

**Comment by jacobvosmaer** [comment]
> So I suppose it must be /usr/local/etc/default/gitlab on BSD family of operating systems.

Yes something like that would be more natural I agree. But I am not sure how to fix that in the init script.

Wouldn't you want a completely different init script that just uses rc.subr anyway? That would be much nicer on FreeBSD now that I think about it.

**Comment by NuLL3rr0r** [comment]
Sorry, my mistake. I thought /etc/default/gitlab is a configuration file that you are mentioning. Yes I already have that file. In fact as part of my installation or upgrade process I always do:
$ sudo cp lib/support/init.d/gitlab /usr/local/etc/rc.d/gitlab

> You can just write this in /etc/default/gitlab:
>    shell_path=/usr/local/bin/bash

Yes, the first thing I did was that. But, until now (GitLab < 7.12.0) I enjoyed installing it like Linux users and it worked out of the box.

> Wouldn't you want a completely different init script that just uses rc.subr anyway? That would be much nicer on FreeBSD now that I think about it.

I never thought of that. So, if it's OK with you I can write one in my spare time since the current one is well documented and contribute it back to be merged.

**Comment by jacobvosmaer** [comment]
This somehow got snowed under in my inbox.

Having had to touch the init script again recently, I think life could be a lot easier on FreeBSD with a native version. It is not something I can actively work on however.

At this point GitLab consists of 4 daemons: unicorn, sidekiq, mail_room and gitlab-git-http-server. I still do not really understand what is the canonical way to manage such a group in FreeBSD, or any other init system for that matter.

**Comment by NuLL3rr0r** [comment]
OK, then I start working on it. But I won't promise a deadline since I'll try working on it on my spare time. I'll report back if I have any success with it.

**Comment by jvanbaarsen** [comment]
This merge request has been closed because a request for more information has not been reacted to for more than 2 weeks. If you respond and conform to the merge request guidelines in our [contributing guidelines](https://gitlab.com/gitlab-org/gitlab-ce/blob/master/CONTRIBUTING.md#pull-requests) we will reopen this merge request.
